pennylane/qnodes/base.py

```diff
@@ -297,10 +303,7 @@ def set_trainable_args(self, arg_indices):
                 f"Argument index not available. QNode has at most {self.func.n_pos} arguments."
             )
 
-        if min(arg_indices) < 0:
-            raise ValueError("Argument indices must be positive integers")
-
-        if not all(isinstance(i, int) for i in arg_indices):
+        if not all(isinstance(i, int) and i >= 0 for i in arg_indices):
```

Comment:
Probably not a big deal if we don't have a lot of parameters, but could maybe get slightly quicker check by using `any` rather than `all`?
```
if any(not isinstance(i, int) or i<0 for i in arg_indices):
```

Reply:
```suggestion
        if any(not isinstance(i, int) or i < 0 for i in arg_indices):
```